lib/parse.js

```diff
@@ -49,7 +49,13 @@ var isoSentinel = 'utf8=%26%2310003%3B'; // encodeURIComponent('&#10003;')
 var charsetSentinel = 'utf8=%E2%9C%93'; // encodeURIComponent('✓')
 
 var parseValues = function parseQueryStringValues(str, options) {
-    var obj = {};
+    // This creates a temporary object to hold parsed query string values.
+    // We need to use a plain object in case the global Object prototype is frozen, otherwise we could encounter a TypeError here if the
+    // query string contains properties that exist on the Object prototype.
+    // If the "allowPrototypes" option is set to false (default), then any query properties that _would_ overwrite properties on the Object
+    // prototype will be stripped by the parseKeys function before being returned.
+    var obj = Object.create(null);
```

Comment:
```suggestion
    var obj = { __proto__: null };
```
instead of the comments, how about a test?

Reply:
Sure, I can add a test!

Regarding this comment:

> (if Object.prototype is frozen, violating the spec, then Object.create surely might be deleted)

I'm happy to change it. If this is a concern, do you think other existing usages of `Object.create(null)` should be updated as well?

Reply:
Yes, but let's not do that in this PR. I can do it separately.

Reply:
Okay, I added a test in 64d88946b0943e908e86e52c41c739a43418ffc2, ready for another pass.

I had to get a little creative, but I think it's sufficient. Let me know what you think, thanks!